source/jquery.flot.legend.js

```diff
@@ -61,26 +61,32 @@
                 iconHtml += getEntryIconHtml(shape);
             }
             // bars
-            if (entry.options.bars.show) {
+            else if (entry.options.bars.show) {
                 shape.name = 'bar';
                 shape.fillColor = entry.color;
                 iconHtml += getEntryIconHtml(shape);
             }
             // lines
-            if (entry.options.lines.show && !entry.options.lines.fill) {
+            else if (entry.options.lines.show && !entry.options.lines.fill) {
                 shape.name = 'line';
                 shape.strokeColor = entry.color;
                 shape.strokeWidth = entry.options.lines.lineWidth;
                 iconHtml += getEntryIconHtml(shape);
             }
             // points
-            if (entry.options.points.show) {
+            else if (entry.options.points.show) {
                 shape.name = entry.options.points.symbol;
                 shape.strokeColor = entry.color;
                 shape.fillColor = entry.options.points.fillColor;
                 shape.strokeWidth = entry.options.points.lineWidth;
                 iconHtml += getEntryIconHtml(shape);
             }
+            // Other
+            else {
+                shape.name = 'box';
+                shape.fillColor = entry.color;
+                iconHtml += getEntryIconHtml(shape);
```

Comment:
It looks like what you're trying to say is if we don't have a known shape based on the series type, we should instead just draw a box with color for each entry in the legend. Instead, would it be better to offer a chance for a plugin to provide their own HTML for a legend entry? Here we could simply look for a known function name (which we would publish as part of the legend API docs), like 'getCustomIcomHtml()'. Then, the pie chart plugin would simply define this method on its options.